crates/trigger/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+use std::{convert::TryFrom, error::Error, path::PathBuf};
+
+use anyhow::{Context, Result};
+use async_trait::async_trait;
+use spin_engine::{Builder, ExecutionContext, ExecutionContextConfiguration};
+use spin_manifest::{Application, ApplicationTrigger, ComponentMap, CoreComponent, TriggerConfig};
+
+#[async_trait]
+pub trait Trigger: Sized {
+    type ContextData: Default + 'static;
+    type Config;
+    type ComponentConfig;
+    type RunConfig;
+
+    fn new(
+        execution_context: ExecutionContext<Self::ContextData>,
+        config: Self::Config,
+        component_configs: ComponentMap<Self::ComponentConfig>,
```

Comment:
we can simply pass `Application<CoreComponent>` to new, instead of `config` and `component_configs`. Is this design intentionally trying to decouple Application and Trigger?

Reply:
Yes it's intentionally decoupling, but it also avoids needing to `clone` the `Application`, which seems to be part of the issue in #413.

Reply:
it seems like Http trigger will need application to build routers